src/components/FlatmapVuer.vue

```diff
@@ -1178,10 +1178,14 @@ export default {
     processTaxon: function (flatmapAPI, taxonIdentifiers) {
       this.taxonConnectivity.length = 0
       taxonIdentifiers.forEach((taxon) => {
-        findTaxonomyLabel(flatmapAPI, taxon).then((value) => {
-          const item = { taxon, label: value }
-          this.taxonConnectivity.push(item)
-        })
+        this.mapImp.queryLabels(taxon).then((entityLabels) => {
```

Comment:
This can api can take a lost of taxons now, please try that.

Reply:
@alan-wu this is updated now.